Source/Core/DolphinQt2/Config/LoggerWidget.cpp

```diff
@@ -0,0 +1,435 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt2/Config/LoggerWidget.h"
+
+#include <QCheckBox>
+#include <QComboBox>
+#include <QEvent>
+#include <QFont>
+#include <QFontDatabase>
+#include <QGridLayout>
+#include <QGroupBox>
+#include <QListWidget>
+#include <QPushButton>
+#include <QRadioButton>
+#include <QScrollBar>
+#include <QSettings>
+#include <QTabWidget>
+#include <QTextEdit>
+#include <QTimer>
+#include <QVBoxLayout>
+
+#include <algorithm>
+
+#include "Common/FileUtil.h"
+#include "Core/ConfigManager.h"
+#include "DolphinQt2/Settings.h"
+
+// Delay in ms between calls of UpdateLog()
+constexpr int UPDATE_LOG_DELAY = 100;
+// Maximum lines to process at a time
+constexpr int MAX_LOG_LINES = 200;
+// Timestamp length
+constexpr int TIMESTAMP_LENGTH = 10;
+
+LoggerWidget::LoggerWidget(QWidget* parent)
+    : QDockWidget(tr("Logging"), parent), m_timer(new QTimer(this))
+{
+  setAllowedAreas(Qt::AllDockWidgetAreas);
+
+  CreateWidgets();
+  CreateMainLayout();
+  LoadSettings();
+
+  ConnectWidgets();
+  OnTabVisibilityChanged();
+
+  LogManager::GetInstance()->RegisterListener(LogListener::LOG_WINDOW_LISTENER, this);
+
+  connect(m_timer, &QTimer::timeout, this, &LoggerWidget::UpdateLog);
+  connect(&Settings::Instance(), &Settings::LogVisibilityChanged, this,
+          &LoggerWidget::OnTabVisibilityChanged);
+  connect(&Settings::Instance(), &Settings::LogConfigVisibilityChanged, this,
+          &LoggerWidget::OnTabVisibilityChanged);
+
+  m_timer->start(UPDATE_LOG_DELAY);
+
+  QSettings settings;
+
+  restoreGeometry(settings.value(QStringLiteral("logger/geometry")).toByteArray());
+  setFloating(settings.value(QStringLiteral("logger/floating")).toBool());
+
+  installEventFilter(this);
+}
+
+LoggerWidget::~LoggerWidget()
+{
+  SaveSettings();
+
+  LogManager::GetInstance()->EnableListener(LogListener::LOG_WINDOW_LISTENER, false);
```

Comment:
This is a bug (see #5772 for the DolphinWX fix and slightly more detail) that will always save "WriteToWindow = False". The correct logic here is

```c++
LogManager::GetInstance()->RegisterListener(LogListener::LOG_WINDOW_LISTENER, nullptr);
```